### PR TITLE
Ragaaf (RAG assessment annotation free) 

### PR DESCRIPTION
**Pull request by adkakne**
## Description

Adding annotation free RAG evaluation feature using LLM-as-a-judge methodology. 

## Issues

`n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds new functionality)

## Dependencies

jsonlines, openai

## Tests

Added `./tests/test_ragaaf.py` for testing.

### CODE REVIEW

File: tests/test_ragaaf.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+host_ip = os.getenv("host_ip", "localhost")
+port = os.getenv("port", "8008")
+
+
+class TestRagasMetric(unittest.TestCase):
+
+    # @unittest.skip("need pass localhost id")
+    def test_ragas(self):
+
+        dataset = "explodinggradients/ragas-wikiqa"
```
Comment (lkk12014402) [inline tests/test_ragaaf.py:20]: shall we give a simple case to run the ragaaf like ragas? https://github.com/opea-project/GenAIEval/blob/main/tests/test_ragas.py#L34

Comment (adkakne) [reply tests/test_ragaaf.py]: Hi Kaokao, Thank you for sharing this requirement. I added support for unit test cases in our code. Thank you.

**Comment by adkakne** [comment]
Test is working with local endpoint for `mistralai/Mixtral-8x7B-Instruct-v0.1` sharded on 8 HPUs. 
image

**Comment by adkakne** [comment]
Hi @XinyuYe-Intel @lkk12014402 @minmin-intel, 

After your review, I made updates to RAGAAF codebase, our final offering is - 
1. Data - can accept unit test cases, local datasets or benchmarking datasets 
2. Models - can accept all OpenAI models or a endpoint. 
3. Hardware - endpoint can run on HPU/GPU/CPU, local testing supports GPU/CPU, OpenAI testing uses their remote server. 
4. Metrics - we offer factualness, correctness, relevance and readability. User can add their own metrics easily. 

Thank you for your final review and your insights for improvements in the code.

**Review by lkk12014402 — APPROVED** [review]

**Comment by lkk12014402** [comment]
LGTM

**Review by XinyuYe-Intel — APPROVED** [review]

**Pull request merged**